### ISSUE TITLE
Backport Zhar multiplayer fixes from @pionere

### ISSUE THREAD

**Issue by AJenbo** [diasurgical/devilutionX#2995]
@pionere appears to have made the Zhar quest work in multiplayer:
https://github.com/pionere/devilutionX/commit/16a4ad88952be742e5b692c75725f60717fa180b

It would be good to have this back ported so that can be one step closer to having support for all content in multiplayer.

**Comment by Chance4us** [comment]
@pionere has made a lot of great changes in his repository. Maybe he can support us?
It would be really great to experience the game in its full range of playability.

**Comment by pionere** [comment]
We already discussed this with @AJenbo. There are too many differences (coding style, goals, etc...), which makes direct collaboration impossible. You can reach me on Discord if you have specific questions...

**Comment by julealgon** [comment]
@pionere do you have a list of things you've changed thus far? I'm curious to know.

**Comment by AJenbo** [comment]
> @pionere do you have a list of things you've changed thus far? I'm curious to know.

Maybe Discord would be a better place for this conversation as it's not related to the issue, pionere goes by pypy on Discord.

**Comment by pionere** [comment]
> @pionere do you have a list of things you've changed thus far? I'm curious to know.

No, I don't have a list and it would be too much work to create one at this point.
